sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -436,6 +436,14 @@ object StructType extends AbstractDataType {
    */
   def fromDDL(ddl: String): StructType = CatalystSqlParser.parseTableSchema(ddl)
 
+  /**
+   * Converts a value of StructType to a string in DDL format. For example:
+   * `StructType(Seq(StructField("a", IntegerType)))` should be converted to `a int`
+   */
+  def toDDL(struct: StructType): String = {
+    struct.map(field => s"${quoteIdentifier(field.name)} ${field.dataType.sql}").mkString(",")
```

Comment:
Can this also handle the special character ('\n', '\t', '\', ...) that needs an escape?

Reply:
We use similar code in `SHOW CREATE TABLE`: https://github.com/apache/spark/blob/bac50aa37168a7612702a4503750a78ed5d59c78/sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala#L999